crates/chain/src/indexed_tx_graph.rs

```diff
@@ -224,20 +224,26 @@ where
     /// Irrelevant transactions in `txs` will be ignored.
     pub fn apply_block_relevant(
         &mut self,
-        block: Block,
+        block: &Block,
         height: u32,
     ) -> ChangeSet<A, I::ChangeSet> {
         let block_id = BlockId {
             hash: block.block_hash(),
             height,
         };
-        let txs = block.txdata.iter().enumerate().map(|(tx_pos, tx)| {
-            (
-                tx,
-                core::iter::once(A::from_block_position(&block, block_id, tx_pos)),
-            )
-        });
-        self.batch_insert_relevant(txs)
+        let mut changeset = ChangeSet::<A, I::ChangeSet>::default();
+        for (tx_pos, tx) in block.txdata.iter().enumerate() {
+            changeset.indexer.append(self.index.index_tx(tx));
+            if self.index.is_tx_relevant(tx) {
+                let txid = tx.txid();
+                let anchor = A::from_block_position(block, block_id, tx_pos);
```

Comment:
It's because there are 3 inputs and doing `A: From<(&Block, BlockId, usize)>` is confusing and we won't know what the `usize` is for.

An alternative would be to have:

```rust
/// Parameters for constructing an [`Anchor`].
pub struct AnchorParams<'b> {
    pub block: &'b Block,
    pub block_id: BlockId,
    pub tx_pos: usize,
}

fn some_method<A: Anchor + From<AnchorParams>>() { todo!() }
```

What do you think? (`AnchorParams` will be nicely documented of course!)

Reply:
I've created an issue to further this discussion: https://github.com/bitcoindevkit/bdk/issues/1266